srf_generation/velocity_model_generation/q_model_generation/generate_qp_qs.py

```diff
@@ -129,6 +130,45 @@ def generate_q_file(
         q_data.save(output)
 
 
+def add_basins(vm_dir, vm_params, outfile_prefix):
+    """
+    Go back and modify basin regions in qp/qs model.
+    """
+    vs = open(join(vm_dir, "vs3dfile.s"), "rb")
+    qp = open(f"{outfile_prefix}.qp", "rb+")
+    qs = open(f"{outfile_prefix}.qs", "rb+")
+    basin_mask = open(join(vm_dir, "in_basin_mask.b"), "rb")
+
+    nx = vm_params["nx"]
+    bytes_x = nx * 4
+    # used to skip binary files where there are no basins
+    seek = 0
+    # dimensionality isn't important, could do zx slices too
+    for _ in range(vm_params["ny"]):
+        for _ in range(vm_params["nz"]):
+            # float array of ints in file, probably the basin index + 1
+            basin_x = np.fromfile(basin_mask, dtype="f4", count=nx) > 0
+            if not basin_x.any():
+                seek += bytes_x
```

Comment:
Is it an issue if we do the seeking here? I'd see that as easier to read rather than accumulating the seeks?

Reply:
not as nice computationally but I guess it doesn't matter